### PR TITLE
Refactoring of proof manager initialization

### PR DESCRIPTION
**Pull request by ajreynol**
No longer takes a backwards reference to SmtEngine.

Also takes minor changes to proof post-processor from proof-new.

### CODE REVIEW

File: src/smt/env.h [hunk 1]
```diff
@@ -56,6 +57,7 @@ class TrustSubstitutionMap;
 class Env
 {
   friend class SmtEngine;
+  friend class smt::PfManager;
```
Comment (aniemetz) [inline src/smt/env.h:60]: I don't think this is necessary?

Comment (ajreynol) [reply src/smt/env.h:60]: We need this to call setProofNodeManager actually.

The reason for this: (1) Env has to be constructed first, (2) we don't know if proofs are enabled until after setDefaults runs, (3) Env needs pointer to the ProofNodeManager if/when proofs are enabled.

At the moment, this means we need a call to private initialization of setProofNodeManager in the constructor of PfManager.  I'm not sure if there is a cleaner way to do this.

Comment (aniemetz) [reply src/smt/env.h:60]: I think there's no really good reason to have `Env::setProofNodeManager` private, is there?

Comment (aniemetz) [reply src/smt/env.h:60]: After discussing this offline, we decided that this is fine. We only want `PfManager` to be able to call `Env::setProofNodeManager`, that's why it's private.

**Review by aniemetz — APPROVED** [review]

**Pull request merged**